doc/reference/kernel/other/event_manager.rst

```diff
@@ -0,0 +1,359 @@
+
+.. code-block:: c
+
+   #include <event_manager.h>
+
+   struct sample_event {
+	   struct event_header header;
+
+	   /* Custom data fields. */
+	   int8_t value1;
+	   int16_t value2;
+	   int32_t value3;
+   };
+
+   EVENT_TYPE_DECLARE(sample_event);
+
+In some use cases, the length of the data associated with an event may vary.
+You can use the :c:macro:`EVENT_TYPE_DYNDATA_DECLARE` macro instead of :c:macro:`EVENT_TYPE_DECLARE` to declare an event type with variable data size.
+In such case, add the data with the variable size as the last member of the event structure.
+For example, you can add the variable size data to a previously defined event by applying the following change to the code:
+
+.. code-block:: c
+
+   #include <event_manager.h>
+
+   struct sample_event {
+	   struct event_header header;
+
+	   /* Custom data fields. */
+	   int8_t value1;
+	   int16_t value2;
+	   int32_t value3;
+	   struct event_dyndata dyndata;
+   };
+
+   EVENT_TYPE_DYNDATA_DECLARE(sample_event);
+
+In this example, the :c:struct:`event_dyndata` contains the following information:
+
+* A zero-length array that is used as a buffer with variable size (:c:member:`event_dyndata.data`).
+* A number representing the size of the buffer (:c:member:`event_dyndata.size`).
+
+Create a source file
+--------------------
+
+To create a source file for the event type you defined in the header file:
+
+1. Include the header file for the new event type in your source file.
+#. Define the event type with the :c:macro:`EVENT_TYPE_DEFINE` macro.
+   Passing the name of the event type as declared in the header and the additional parameters.
+   For example, you can provide a function that fills a buffer with a string version of the event data (used for logging).
+
+The following code example shows a source file for the event type ``sample_event``:
+
+.. code-block:: c
+
+   #include "sample_event.h"
+
+   static int log_sample_event(const struct event_header *eh, char *buf,
+			       size_t buf_len)
+   {
+	   struct sample_event *event = cast_sample_event(eh);
+
+	   return snprintf(buf, buf_len, "val1=%d val2=%d val3=%d", event->value1,
+			   event->value2, event->value3);
+   }
+
+   EVENT_TYPE_DEFINE(sample_event,	/* Unique event name. */
+		     true,		/* Event logged by default. */
+		     log_sample_event,	/* Function logging event data. */
+		     NULL);		/* No event info provided. */
+
+Submitting an event
+===================
+
+To submit an event of a given type, for example ``sample_event``:
+
+1. Allocate the event by calling the function with the name *new_event_type_name*.
+   For example, ``new_sample_event()``.
+#. Write values to the data fields.
+#. Use :c:macro:`EVENT_SUBMIT` to submit the event.
+
+The following code example shows how to create and submit an event of type ``sample_event`` that has three data fields:
+
+.. code-block:: c
+
+	/* Allocate event. */
+	struct sample_event *event = new_sample_event();
```

Comment:
is this dynamic allocation?

Reply:
Yes - there are hooks (weak linkage) that allows user to pass own allocator functions. by default it's k_malloc.